### ISSUE TITLE
keyword labelrotation is not recognized

### ISSUE THREAD

**Issue by ggrrll** [matplotlib/matplotlib#8122]
### Bug report

**Bug summary**
in [ax.tick_params](http://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.tick_params.html), I get this error 

```keyword labelrotation is not recognized```
**Code for reproduction**

```
ax.tick_params(labelsize=10,labelrotation=40)
```

**Actual outcome**

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-186-fd6800d79e43> in <module>()
     12 
     13 
---> 14 ax.tick_params(labelsize=10,labelrotation=40)
     15 
     16 ax.set_xlabel('Fat',fontsize=16)

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/mpl_toolkits/mplot3d/axes3d.py in tick_params(self, axis, **kwargs)
   1395             This function was added, but not tested. Please report any bugs.
   1396         """
-> 1397         Axes.tick_params(self, axis, **kwargs)
   1398         if axis in ['z', 'both'] :
   1399             zkw = dict(kwargs)

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/axes/_base.py in tick_params(self, axis, **kwargs)
   2711             xkw.pop('labelleft', None)
   2712             xkw.pop('labelright', None)
-> 2713             self.xaxis.set_tick_params(**xkw)
   2714         if axis in ['y', 'both']:
   2715             ykw = dict(kwargs)

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/axis.py in set_tick_params(self, which, reset, **kw)
    789         if which == 'minor' or which == 'both':
    790             dicts.append(self._minor_tick_kw)
--> 791         kwtrans = self._translate_tick_kw(kw, to_init_kw=True)
    792         for d in dicts:
    793             if reset:

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/axis.py in _translate_tick_kw(kw, to_init_kw)
    860                     raise ValueError(
    861                         "keyword %s is not recognized; valid keywords are %s"
--> 862                         % (key, kwkeys))
    863             kwtrans.update(kw)
    864         else:

ValueError: keyword labelrotation is not recognized; valid keywords are ['size', 'width', 'color', 'tickdir', 'pad', 'labelsize', 'labelcolor', 'zorder', 'gridOn', 'tick1On', 'tick2On', 'label1On', 'label2On', 'length', 'direction', 'left', 'bottom', 'right', 'top', 'labelleft', 'labelbottom', 'labelright', 'labeltop']
```

**Comment by ggrrll** [comment]
(taken from
 [this doc page](http://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.tick_params.html) )

**Comment by ggrrll** [comment]
- macOS 10.12.3
- Name: matplotlib / Version: 2.0.0

**Comment by phobson** [comment]
It appears you need to install from the current git master to use the `labelrotation` kwarg.

Compare [v2.0 docs](http://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.tick_params.html?highlight=tick_params#matplotlib.axes.Axes.tick_params) vs. [master docs](http://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.tick_params.html?highlight=tick_params#matplotlib.axes.Axes.tick_params)

[](url)

**Comment by QuLogic** [comment]
Yep, you're looking at development docs, not 2.0.0 docs.

**Comment by ggrrll** [comment]
hi, 

when I try to install the dev, somehow I managed to skrew up all my python settings... :(

if you have any idea on how to solve [my issue](https://github.com/ipython/ipython/issues/10323)
...thank you very much in advance!

**Comment by tdpetrou** [comment]
I just installed 2.1 and `labelrotation` does not work. `rotation` does however. Also it would be nice if `ax.tick_params` did horizontal alignment as well.

**Comment by tacaswell** [comment]
The docs do in fact say `labelrotation` should work, but it does not 

http://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.tick_params.html

**Comment by tacaswell** [comment]
Not sure if the correct fix is to make the docs match the code or the code match the docs though.

**Comment by jklymak** [comment]
Rotation is used elsewhere.  I prefer kwargs stay the same between calls.  Right now I can never remember if it’s location or loc.

**Comment by WeatherGod** [comment]
as a rule of thumb, yes, I agree. What we do need to watch out for are name
collisions. This is why we have `color`, `edgecolor`, 'linecolor`, ..... If
there is something else that `rotation` could refer to in this context,
then labelrotation makes it clear what should be rotated

**Comment by tacaswell** [comment]
Is there a way to rotate the _ticks_ ?

**Comment by jklymak** [comment]
This is closed via #9633 I believe
